File: AuditTracking.Api/Entities/RectificationVerification.cs
using System.ComponentModel.DataAnnotations;

namespace AuditTracking.Api.Entities;

/// <summary>
/// 整改验证记录。
/// </summary>
public sealed class RectificationVerification
{
    /// <summary>
    /// 主键。
    /// </summary>
    public int Id { get; set; }

    /// <summary>
    /// 所属审计问题 ID。
    /// </summary>
    [Required]
    public int AuditIssueId { get; set; }

    /// <summary>
    /// 所属整改措施 ID。
    /// </summary>
    [Required]
    public int CorrectiveActionId { get; set; }

    /// <summary>
    /// 验证编号。
    /// </summary>
    [Required]
    [MaxLength(50)]
    public string VerificationNo { get; set; } = string.Empty;

    /// <summary>
    /// 验证结果：
    /// Passed、Failed、NeedMoreEvidence。
    /// </summary>
    [Required]
    [MaxLength(30)]
    public string VerificationResult { get; set; } = string.Empty;

    /// <summary>
    /// 验证意见。
    /// </summary>
    [Required]
    [MaxLength(4000)]
    public string VerificationComment { get; set; } = string.Empty;

    /// <summary>
    /// 验证人。
    /// </summary>
    [Required]
    [MaxLength(100)]
    public string Verifier { get; set; } = string.Empty;

    /// <summary>
    /// 验证时间。
    /// </summary>
    public DateTime VerifiedAt { get; set; }

    /// <summary>
    /// 是否通过。
    /// </summary>
    public bool IsPassed { get; set; }

    /// <summary>
    /// 创建时间。
    /// </summary>
    public DateTime CreatedAt { get; set; } = DateTime.Now;

    /// <summary>
    /// 创建人。
    /// </summary>
    [Required]
    [MaxLength(100)]
    public string CreatedBy { get; set; } = "System";

    /// <summary>
    /// 最后修改时间。
    /// </summary>
    public DateTime? UpdatedAt { get; set; }

    /// <summary>
    /// 最后修改人。
    /// </summary>
    [MaxLength(100)]
    public string? UpdatedBy { get; set; }

    /// <summary>
    /// 是否已软删除。
    /// </summary>
    public bool IsDeleted { get; set; }

    /// <summary>
    /// 删除时间。
    /// </summary>
    public DateTime? DeletedAt { get; set; }

    /// <summary>
    /// 删除人。
    /// </summary>
    [MaxLength(100)]
    public string? DeletedBy { get; set; }

    /// <summary>
    /// 所属审计问题。
    /// </summary>
    public AuditIssue AuditIssue { get; set; } = null!;

    /// <summary>
    /// 所属整改措施。
    /// </summary>
    public CorrectiveAction CorrectiveAction { get; set; } = null!;
}

File: audit-tracking-web/src/types/rectificationVerification.ts
export type VerificationResult = 'Passed' | 'Failed' | 'NeedMoreEvidence'

export interface RectificationVerification {
  id: number
  auditIssueId: number
  correctiveActionId: number
  verificationNo: string
  verificationResult: VerificationResult
  verificationComment: string
  verifier: string
  verifiedAt: string
  isPassed: boolean
  createdAt: string
  createdBy: string
  updatedAt: string | null
  updatedBy: string | null
  isDeleted?: boolean
  deletedAt?: string | null
  deletedBy?: string | null
}

export interface RectificationVerificationListItem extends RectificationVerification {}

export interface RectificationVerificationDetail extends RectificationVerification {
  issueNo: string
  issueTitle: string
  actionNo: string
  actionDescription: string
  auditPlanId: number
}

export interface RectificationVerificationQuery {
  keyword?: string
  auditIssueId?: number | null
  correctiveActionId?: number | null
  verificationResult?: VerificationResult | ''
  isPassed?: boolean | null
  verifiedDateStart?: string | null
  verifiedDateEnd?: string | null
  page: number
  pageSize: number
}

export interface CreateRectificationVerificationRequest {
  auditIssueId: number
  correctiveActionId: number
  verificationNo: string
  verificationResult: VerificationResult
  verificationComment: string
  verifier: string
  verifiedAt?: string | null
}

export interface UpdateRectificationVerificationRequest {
  verificationResult: VerificationResult
  verificationComment: string
  verifier: string
  verifiedAt?: string | null
}

export interface RectificationVerificationOperationLog {
  id: number
  rectificationVerificationId: number
  verificationNo: string
  operationType: string
  beforeData: string | null
  afterData: string | null
  operator: string
  remark: string | null
  createdAt: string
}

export interface AuditIssueOption {
  id: number
  issueNo: string
  title: string
}

export interface CorrectiveActionOption {
  id: number
  auditIssueId: number
  actionNo: string
  actionDescription: string
  status: string
}

export interface PagedResult<T> {
  items: T[]
  page: number
  pageSize: number
  total: number
  totalPages: number
}
